kyo-offheap/native/src/main/java/lang/foreign/MemorySegment.scala

```diff
@@ -0,0 +1,117 @@
+package java.lang.foreign
+
+import scala.scalanative.unsafe._
+import scala.scalanative.libc.stdlib.malloc
+import scala.scalanative.libc.string.memcpy
+import scala.scalanative.unsigned._
+
+/** A minimal stub for Java's MemorySegment using Scala Native pointers.
+  *
+  * This implementation wraps a pointer along with the allocated size in bytes.
+  * It also provides basic support for slicing and for reading/writing primitive values,
+  * which are used by the Layout instances in the shared code.
+  */
+final class MemorySegment private (ptr: Ptr[Byte], val byteSize: Long) {
+  /** Creates a new MemorySegment that is a slice of this segment.
+    *
+    * @param offset The offset (in bytes) at which the slice begins.
+    * @param newSize The size (in bytes) of the new segment.
+    * @return A new MemorySegment representing the slice.
+    */
+  def asSlice(offset: Long, newSize: Long): MemorySegment = {
+    if (offset < 0 || newSize < 0 || offset + newSize > byteSize)
+      throw new IllegalArgumentException("Invalid slice parameters")
+    new MemorySegment(ptr + offset, newSize)
+  }
+
+  /** Reads a value from memory using the provided layout.
+    */
+  def get(layout: ValueLayout.ofByte, offset: Long): Byte = {
+    !(ptr + offset).cast[Byte]
+  }
+  def get(layout: ValueLayout.ofShort, offset: Long): Short = {
+    !(ptr + offset).cast[Short]
+  }
+  def get(layout: ValueLayout.ofInt, offset: Long): Int = {
+    !(ptr + offset).cast[Int]
+  }
+  def get(layout: ValueLayout.ofLong, offset: Long): Long = {
+    !(ptr + offset).cast[Long]
+  }
+  def get(layout: ValueLayout.ofFloat, offset: Long): Float = {
+    !(ptr + offset).cast[Float]
+  }
+  def get(layout: ValueLayout.ofDouble, offset: Long): Double = {
+    !(ptr + offset).cast[Double]
+  }
+  def get(layout: ValueLayout, offset: Long): MemorySegment =
```

Comment:
Please look carefully at my original comment https://github.com/getkyo/kyo/pull/1072#discussion_r1953547609
```suggestion
  def get(layout: AddressLayout, offset: Long): MemorySegment =
```